src/util/scale/getNiceTickValues.ts

```diff
@@ -45,12 +45,19 @@ export const getFormatStep = (roughStep: Decimal, allowDecimals: boolean, correc
   const digitCountValue = new Decimal(10).pow(digitCount);
   const stepRatio = roughStep.div(digitCountValue);
   // When an integer and a float multiplied, the accuracy of result may be wrong
-  const stepRatioScale = digitCount !== 1 ? 0.05 : 0.1;
-  const amendStepRatio = new Decimal(Math.ceil(stepRatio.div(stepRatioScale).toNumber()))
-    .add(correctionFactor)
-    .mul(stepRatioScale);
+  let stepRatioScale = digitCount !== 1 ? 0.05 : 0.1;
 
-  const formatStep = amendStepRatio.mul(digitCountValue);
+  let formatStep: Decimal;
+
+  do {
```

Comment:
this should be in a regular while loop I think? In a do-while 0.05 will always get decremented to 0.04 before the check